### PR TITLE
python3Packages.phylophlan: init at 3.1.1

### LINKED ISSUE #343368: Package request: phylophlan
**Project description**

Precise phylogenetic analysis of microbial isolates and genomes from metagenomes

**Metadata**

* homepage URL: https://github.com/biobakery/phylophlan
* source URL: https://github.com/biobakery/phylophlan
* license: mit
* platforms: all

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by theobori**
## Description of changes
Fixes #343368

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ph/phylophlan/package.nix [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  fetchFromGitHub,
+  python3,
+  raxml,
+  mafft,
+  trimal,
+  blast,
+  diamond,
+}:
+let
+  python3Packages = python3.pkgs;
```
Comment (Pandapip1) [inline pkgs/by-name/ph/phylophlan/package.nix:12]:
```suggestion
  python3,
  python3Packages,
  raxml,
  mafft,
  trimal,
  blast,
  diamond,
}:
let
```

File: pkgs/by-name/ph/phylophlan/package.nix [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  finalAttrs = {
+    pname = "phylophlan";
+    version = "3.1.1";
+    pyproject = true;
+
+    src = fetchFromGitHub {
+      owner = "biobakery";
+      repo = "phylophlan";
+      rev = "${finalAttrs.version}";
```
Comment (Pandapip1) [inline pkgs/by-name/ph/phylophlan/package.nix:21]:
```suggestion
      rev = finalAttrs.version;
```

File: pkgs/by-name/ph/phylophlan/package.nix [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    postInstall = ''
+      # Not revelant in this context
+      rm -f $out/bin/phylophlan_write_default_configs.sh
+    '';
+
+    meta = {
+      homepage = "https://github.com/biobakery/phylophlan";
+      description = "Precise phylogenetic analysis of microbial isolates and genomes from metagenomes";
+      changelog = "https://github.com/biobakery/phylophlan/releases/tag/${finalAttrs.version}";
```
Comment (Pandapip1) [inline pkgs/by-name/ph/phylophlan/package.nix:59]: Please consider


```suggestion
      changelog = "https://github.com/biobakery/phylophlan/releases";
```
https://github.com/NixOS/nixpkgs/issues/336454

File: pkgs/by-name/ph/phylophlan/package.nix [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      license = lib.licenses.mit;
+      maintainers = with lib.maintainers; [ theobori ];
+      platforms = lib.platforms.unix;
```
Comment (Pandapip1) [inline pkgs/by-name/ph/phylophlan/package.nix:62]: Not needed; automatically added by `buildPythonAppplication`.


```suggestion
```

**Review by Pandapip1 — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ph/phylophlan/package.nix (hunk 3 above)
Comment (theobori) [reply pkgs/by-name/ph/phylophlan/package.nix]: Yep, it is probably better to set a non-parameterized url.

**Comment by Pandapip1** [comment]
Result of `nixpkgs-review pr 343699` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>phylophlan</li>
    <li>phylophlan.dist</li>
  </ul>
</details>

**Review by Pandapip1 — APPROVED** [review]

**Comment by FliegendeWurst** [comment]
@ofborg build python312Packages.phylophlan

**Comment by FliegendeWurst** [comment]
@ofborg build phylophlan

**Review by FliegendeWurst — APPROVED** [review]

**Comment by Pandapip1** [comment]
:|

**Comment by theobori** [comment]
> :|

Yes I deleted my nixpkgs fork :D.
I will recreate a PR for this pacakge when I have time don't worry.

**Comment by theobori** [comment]
The new PR is here https://github.com/NixOS/nixpkgs/pull/356788.

**Pull request closed without merging**